crates/bitwarden-ipc/src/serde_utils.rs

```diff
@@ -0,0 +1,18 @@
+//! Global serialization and deserialization utilities for IPC messages.
+//! This module provides functions to serialize and deserialize IPC messages in one place,
+//! ensuring consistency and reducing code duplication across the IPC crate.
+
+use serde::{Deserialize, Serialize};
+
+pub(crate) type SerializeError = serde_json::Error;
+pub(crate) type DeserializeError = serde_json::Error;
+
+pub(crate) fn to_vec<T: Serialize>(value: &T) -> Result<Vec<u8>, serde_json::Error> {
+    serde_json::to_vec(value)
+}
+
+pub(crate) fn from_slice<T: for<'de> Deserialize<'de>>(
```

Comment:
You should be able to use `DeserializeOwned` which is the same thing and slightly easier to understand: https://docs.rs/serde/latest/serde/de/trait.DeserializeOwned.html

Reply:
You are too quick haha

Reply:
😎